storage_service/locations/api/beta/remple/openapi.py

```diff
@@ -0,0 +1,1573 @@
+                    'schema': {
+                        '$ref':
+                        self._get_create_schema_path(resource_name)
+                    }
+                }
+            }),
+        ])
+
+    def _get_update_request_body(self, resource_name):
+        return OrderedDict([
+            ('description', 'JSON object required to update an existing'
+                            ' {}'.format(resource_name)),
+            ('required', True),
+            ('content', {
+                'application/json': {
+                    'schema': {
+                        '$ref':
+                        self._get_update_schema_path(resource_name)
+                    }
+                }
+            }),
+        ])
+
+    def _get_search_request_body(self, resource_name, examples=None):
+        rsrc_collection_name = self.inflp.plural(resource_name)
+        application_json = OrderedDict([
+            ('schema', {'$ref':
+                        self._get_search_schema_path(resource_name)}),
+        ])
+        if examples:
+            application_json['example'] = examples
+        return OrderedDict([
+            ('description', 'JSON object required to search over all {}'.format(
+                rsrc_collection_name)),
+            ('required', True),
+            ('content', {'application/json': application_json}),
+        ])
+
+    def _get_search_post_request_body(self, resource_name, examples=None):
+        return self._get_search_request_body(resource_name, examples=examples)
+
+    def _get_index_parameters(self, *args):
+        """Return an OpenAPI ``parameters`` object for the "index" action."""
+        return [
+            {'$ref': '#/components/parameters/items_per_page'},
+            {'$ref': '#/components/parameters/page'},
+            {'$ref': '#/components/parameters/order_by_attribute'},
+            {'$ref': '#/components/parameters/order_by_subattribute'},
+            {'$ref': '#/components/parameters/order_by_direction'},
+        ]
+
+    def to_yaml(self, open_api_spec):
+        """Return the input OrderedDict as a YAML string."""
+        return yaml_dump(open_api_spec)
+
+    def to_json(self, open_api_spec):
+        """Return the input OrderedDict as a JSON string."""
+        return json.dumps(open_api_spec)
+
+    def _get_api_info(self):
+        """Return an OrderedDict for the top-level ``info`` attribute."""
+        return OrderedDict([
+            ('version', self.api_version),
+            ('title', self._get_api_title()),
+            ('description', self._get_api_description()),
+        ])
+
+    def get_dflt_server_path(self):
+        return '{}{}'.format(
+            self.path_prefix, self.get_api_version_slug())
+
+    def _get_api_servers(self):
+        """Return a list of OrderedDicts for the top-level ``servers``
+        attribute.
+        """
+        return [
+            OrderedDict([
+                ('url', self.get_dflt_server_path()),
```

Comment:
TODO: should we be able to use info in `django_settings` to specify a true URL here instead of just a path?